microscopic-monks/primal/engine/sprite.py

```diff
@@ -0,0 +1,228 @@
+from pathlib import Path
+from typing import Tuple, Union
+
+from kivy.graphics.context_instructions import PushMatrix, PopMatrix
+from kivy.core.window import Window
+from kivy.graphics import Rectangle, Color
+from kivy.graphics.instructions import RenderContext, InstructionGroup
+from kivy.graphics.context_instructions import Rotate
+from kivy.uix.label import Label
+
+import math
+import json
+
+
+class Text:
+    def __init__(self, text: str, pos: Tuple[float, float], height: float):
+        self.pos = pos
+        self.height = height
+
+        self.label = Label()
+        self.label.font_size = 20
+        self.label.bold = True
+        self.label.text_size = 800, None
+        self.color = Color(1, 1, 1, 1)
+        self.rect = Rectangle(pos=pos)
+        self.set_text(text)
+
+    def set_text(self, text: str):
+        self.label.text = text
+        self.label.texture_update()
+        ratio = self.label.texture_size[0] / self.label.texture_size[1]
+
+        self.rect.size = ratio * self.height, self.height
+        self.rect.texture = self.label.texture
+        self.rect.texture.mag_filter = 'nearest'
+        self.rect.texture.min_filter = 'nearest'
+
+    def set_color(self, color: Tuple[float, float, float, float]):
+        self.color.r = color[0]
+        self.color.g = color[1]
+        self.color.b = color[2]
+        self.color.a = color[3]
+
+    def draw(self, canvas: RenderContext):
+        canvas.add(self.color)
+        canvas.add(self.rect)
+        canvas.add(Color(1, 1, 1, 1))
+
+
+class Sprite:
+    resource_dir = Path('.', 'primal', 'resources').absolute()
+    try:
+        f = open(resource_dir / "testimg.png")
```

Comment:
Using the pathlib open context manager and the path exist method is preferable to a manual open and close.
when it comes to working with the file-system, using pathlib to its fullest is currently the best way, even if you like the ask for permission later model with a try/except.